src/client.c

```diff
@@ -894,9 +896,39 @@ static bool handle_client_work(PgSocket *client, PktHdr *pkt)
 	/* forward the packet */
 	sbuf_prepare_send(sbuf, &client->link->sbuf, pkt->len);
 
-	/* log the query, if needed */
-	if (cf_log_packets)
-		log_pkt_to_buffer(pkt, client);
+	if (cf_log_packets) {
+		if (pkt->type == 'Q' || pkt->type == 'P' || pkt->type == 'B' || pkt->type == 'E') {
+			log_pkt_to_buffer(pkt, client);
+			
+			// Check packet incomplete and assign values to struct
+			if (cf_buffer_incomplete_packets) {
+				if (incomplete_pkt(pkt)) {
+					if ((int)pkt->len > (int)cf_sbuf_len) {
+						client->incomplete_packet_buffer.started = true;
+						client->incomplete_packet_buffer.pkt_data = &pkt->data;
+						client->incomplete_packet_buffer.pkt_len = pkt->len;
+						if (custom_log_frequency_counter >= 500) {
```

Comment:
a little trick to reduce some code, it's basically as fast to do a `% 500` and not reset it to zero, since that's two ops.

so you can remove https://github.com/instacart/pgbouncer/pull/19/files#diff-3e4f237563dfceb182c6252492e7d829c659b7b1b1ba379891339d8247cde885R913-R916 and just replace this with
```c
if (custom_log_frequency_counter % 500 == 0) {
```

Reply:
Yeah so I did it like this since this number could get very large and didn't want that. But now I'm realizing that once we hit the max int it just resets to zero so